lib/python/ray/services.py

```diff
@@ -134,5 +136,5 @@ def start_singlenode_cluster(return_drivers=False, num_objstores=1, num_workers_
     time.sleep(0.5)
     return driver_workers
   else:
-    ray.connect(scheduler_address, objstore_addresses[0], address(IP_ADDRESS, new_worker_port()), print_task_info=print_task_info)
+    ray.connect(scheduler_address, objstore_addresses[0], address(IP_ADDRESS, new_worker_port()), mode=driver_mode)
```

Comment:
SO the thing is you have a variable called `driver_mode` that can be set to something other than `ray.DRIVER_MODE`, which is confusing code-wise. Maybe think of a different name?